app/src/main/java/us/kikin/apps/android/bgplayer/ui/show/ShowHeaderViewHolder.kt

```diff
@@ -1,36 +1,32 @@
 package us.kikin.apps.android.bgplayer.ui.show
 
 import android.view.LayoutInflater
-import android.view.View
 import android.view.ViewGroup
-import android.widget.ImageView
-import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
 import coil.load
-import com.google.android.material.button.MaterialButton
-import us.kikin.apps.android.bgplayer.R
+import us.kikin.apps.android.bgplayer.databinding.ItemShowHeaderBinding
 import us.kikin.apps.android.bgplayer.models.VideoShowModel
 
-class ShowHeaderViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+class ShowHeaderViewHolder(
+    private val binding: ItemShowHeaderBinding
+) : RecyclerView.ViewHolder(binding.root) {
     companion object {
```

Comment:
nit: the Kotlin style guide recommends to have companion objects at the end of the class

Reply:
Depends how "interesting" it is.
If it's just a bunch of constants then yes.
But in this case the companion object is the entry point into the class, so it would be like putting the constructor at the bottom. 
imo it's a good spot for it

Reply:
I would agree if we could have multiple companion object per class and keep the interesting stuff on top and the rest stuff at the bottom. However, this isn't the case, and a consistent coding style is important (we don't want to have a companion object in different place depending on the class). Keeping it at the top just seems like the Java style ported to Kotlin, and while I also prefer it at the top (coming from Java), I would highly suggest to follow the language style guide especially when starting a new project.

Reply:
I think consistency of "important stuff at the top" is better than consistency of "companion below methods".
I know this is against the [official style guide](https://kotlinlang.org/docs/reference/coding-conventions.html), but they don't give any kind of reasoning behind it.
[This is heavily out dated](https://github.com/android/kotlin-guides/issues/11), but I'm gonna have to agree with JW on this matter 🤷